src/scrapers/visa-cal.ts

```diff
@@ -99,10 +99,13 @@ function getAmountData(amountStr: string) {
   } else if (amountStrCln.includes(DOLLAR_CURRENCY_SYMBOL)) {
     amount = -parseFloat(amountStrCln.replace(DOLLAR_CURRENCY_SYMBOL, ''));
     currency = DOLLAR_CURRENCY;
+  } else if (amountStrCln.includes(EURO_CURRENCY_SYMBOL)) {
+    amount = -parseFloat(amountStrCln.replace(EURO_CURRENCY_SYMBOL, ''));
+    currency = EURO_CURRENCY;
   } else {
     const parts = amountStrCln.split(' ');
-    amount = -parseFloat(parts[0]);
-    [, currency] = parts;
+    [currency] = parts;
```

Comment:
I see that you added the euro currency. But why did you changed the fallback behavior and switched places? It was there for a long time now so either it never worked or maybe you are using a different UI layout (maybe left-to-right)?

Reply:
Switching amount and currency worked for me even without the EURO fallback (as opposed to the original code). But maybe its not robust enough. I noticed a [library ](https://www.npmjs.com/package/parse-money) for handling the parsing of currency strings. I think in any case the method should be more robust then assuming the string will be in a certain format. The CC companies are definitely not trying to keep there "API" consistent.